hudi-common/src/main/java/org/apache/hudi/common/table/view/AbstractTableFileSystemView.java

```diff
@@ -244,7 +244,9 @@ private void resetFileGroupsReplaced(HoodieTimeline timeline) {
       }
     });
 
-    Map<HoodieFileGroupId, HoodieInstant> replacedFileGroups = resultStream.collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    // Duplicate key error when insert_overwrite same partition in multi writer, distinct HoodieFileGroupId,If repeated, save instant with the larger timestamp
+    Map<HoodieFileGroupId, HoodieInstant> replacedFileGroups = resultStream.collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue,
```

Comment:
`distinct HoodieFileGroupId,If repeated, save instant with the larger timestamp` -> `keep the instant with greater timestamp when the file group id conflicts`

Reply:
@danny0405  thanks，danny，i have update this comment